src/snowflake/snowpark/dataframe.py

```diff
@@ -725,7 +727,10 @@ def to_local_iterator(
         )
 
     def __copy__(self) -> "DataFrame":
-        return DataFrame(self._session, copy.copy(self._select_statement or self._plan))
+        if isinstance(self._select_statement, MockSelectStatement):
```

Comment:
I think we can remove this if case since `self._select_statement` will always be not None for local testing code path

Reply:
fixed